src/Google/Cache/File.php

```diff
@@ -155,7 +155,7 @@ private function getCacheDir($file, $forWrite)
     // trim the directory separator from the path to prevent double separators
     $storageDir = rtrim($this->path, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $dirHash;
     if ($forWrite && ! is_dir($storageDir)) {
-      if (! mkdir($storageDir, 0700, true)) {
+      if (!mkdir($storageDir, 0777, true)) {
```

Comment:
> CLI cache cleaners will be unable to remove files created by the web interface.

Why do you need to cleanup?

> The web interface will not be able to read files created by a CLI cache warmer.

Is this a big problem?

Think about this scenario.

I have downloaded the credentials file and stored secretly, only available for root.
root process will create a cache which contains the access token from the credentials.

Then any other users should not be able to read that cache.

The correct approach for the cron scenario is to run the process as an appropriate user, not root.